test/AngularServiceGeneratorSpec.coffee

```diff
@@ -0,0 +1,145 @@
+'use strict'
+
+describe 'AngularServiceGenerator', ->
+  EventEmitter = window.testHelpers.events
+  ServiceGenerator = require '../src/AngularServiceGenerator'
+  AngularService = undefined
+  EventAdapter = undefined
+  sandbox = undefined
+  mockService = undefined
+  $rootScope = undefined
+  mockModule = undefined
+  mockReverseModule = undefined
+  mockEvents = undefined
+  mod = undefined
+  modReverse = undefined
+  modNoOpts = undefined
+
+  beforeEach angular.mock.module 'mockapp'
+  beforeEach angular.mock.module require('../src/common').name
+  
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    
+    mockModule = new EventEmitter()
+    mockModule.start = sandbox.stub()
+    sandbox.spy mockModule, 'addListener'
+
+    mockReverseModule =
+      eventEmitter: new EventEmitter()
+      addListener: (callback, eventName) ->
+        @eventEmitter.addListener(eventName, callback)
+    sandbox.spy mockReverseModule, 'addListener'
+    
+    mockEvents =
+      EVENT1: 'event1'
+      EVENT2: 'event2'
+    
+    mod =
+      module: mockModule
+      name: 'mock'
+      options:
+        eventMaps:
+          MockEvents: mockEvents
+          
+    modNoOpts =
+      module: mockModule
+      name: 'mockNoOpts'
+
+    modReverse =
+      module: mockModule
```

Comment:
here I was expecting to see the `mockReverseModule` -- but perhaps all that is really needed is the `flipAddListenerArgs`